### PR TITLE
helm: set useDigest to false

### PR DESCRIPTION
**Pull request by m00lecule**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Purpose of this mr is to align cilium's helm chart `useDigest` values to false. `useDigest` value is disabled for most common containers (agent, operator) but not for those minor ones.

I have discovered those inconsistent values while running cilium on EKS cluster without access to public internet and overriding `*image.repository` with quay.io's own ECR Pull Through Cache. When fetching cilium's images from quay.io to own ECR the image digest is changed by AWS which results in ErrImagePull while enabling hubble-ui without previous default values investigation and making me question why it was working for cilium daemonset in the first place.  

```release-note
helm: aligned all useDigest to false
```

### CODE REVIEW

**Review by gandro — CHANGES_REQUESTED** [review]
Thanks for the PR! Unfortunately I don't think we want to merge this as is.

> useDigest value is disabled for most common containers (agent, operator) but not for those minor ones.
In released Helm charts, we always set `useDigest:true` for all images. If that is not the case, that would be a bug. It is only `false` in the development versions of the Helm chart (i.e. the one currently on `main`)/

The reason why some images (like operator or agent) do have it set to false in the development tree is that those images are built from the tree. Thus we don't want to compute the digest when e.g. testing local builds. All other images however are pulled out-of-tree and thus can have `useDigest` to true.

> I have discovered those inconsistent values while running cilium on EKS cluster without access to public internet and overriding *image.repository with quay.io's own ECR Pull Through Cache. 

Generally speaking, I would expect `*.image.override` to disable the digest check. For example, see here:
https://github.com/cilium/cilium/blob/9b1d3a316a0c91fb3f5ea4aa2bfe534701901493/install/kubernetes/cilium/templates/_helpers.tpl#L20-L26

In other words, `useDigest` is not used, if an override is set. If that is not the case for all images, then that's something we should fix in the override logic. But I think it is better to disable the digest check the override logic, rather than disabling the digest check for non-override users.
